plugins/modules/purefa_pg.py

```diff
@@ -702,31 +943,48 @@ def update_pgroup(module, array):
                 rename = module.params["rename"]
             renamed = True
             if not module.check_mode:
-                try:
-                    array.rename_pgroup(module.params["name"], rename)
-                    module.params["name"] = rename
-                except Exception:
-                    module.fail_json(msg="Rename to {0} failed.".format(rename))
+                if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                    res = array.patch_protection_groups(
+                        names=[module.params["name"]],
+                        protection_group=ProtectionGroup(name=rename),
+                        context_names=[module.params["context"]],
+                    )
+                else:
+                    res = array.patch_protection_groups(
+                        names=[module.params["name"]],
+                        protection_group=ProtectionGroup(name=rename),
+                    )
+                module.params["name"] = rename
+                if res.status_code != 200:
+                    module.fail_json(
+                        msg="Rename to {0} failed. Error: {1}".format(
+                            rename, res.errors[0].message
+                        )
+                    )
         else:
             module.warn(
                 "Rename failed. Protection group {0} already exists in container. Continuing with other changes...".format(
                     module.params["rename"]
                 )
             )
-    if RETENTION_LOCK_VERSION in api_version:
-        arrayv6 = get_array(module)
+    if LooseVersion(RETENTION_LOCK_VERSION) <= LooseVersion(api_version):
         current_pg = list(
-            arrayv6.get_protection_groups(names=[module.params["name"]]).items
+            array.get_protection_groups(names=[module.params["name"]]).items
```

Comment:
context case handling is missing, pls add. Its there in multiple occurrences.
When running in fleet array with context, this needs param to be passed - context_names=[module.params["context"]]